src/components/AutoUpdateTime.js

```diff
@@ -0,0 +1,104 @@
+import {View} from 'react-native';
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import styles from '../styles/styles';
+import DateUtils from '../libs/DateUtils';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import Text from './Text';
+
+const propTypes = {
+    /** Timezone of the user from their personal details */
+    timezone: PropTypes.shape({
+        /** Value of selected timezone */
+        selected: PropTypes.string,
+
+        /** Whether timezone is automatically set */
+        automatic: PropTypes.bool,
+    }).isRequired,
+    ...withLocalizePropTypes,
+};
+
+class AutoUpdateTime extends PureComponent {
+    constructor(props) {
+        super(props);
+        this.getCurrentUserLocalTime = this.getCurrentUserLocalTime.bind(this);
+        this.updateCurrentTime = this.updateCurrentTime.bind(this);
+        this.getTimezoneName = this.getTimezoneName.bind(this);
+        this.state = {
+            currentUserLocalTime: this.getCurrentUserLocalTime(),
+        };
+    }
+
+    componentDidMount() {
+        this.updateCurrentTime();
+    }
+
+    componentDidUpdate() {
+        // Make sure the interval is up to date every time the component updates
+        this.updateCurrentTime();
+    }
+
+    componentWillUnmount() {
+        // clearInterval(this.timer);
+        clearTimeout(this.timer);
+    }
+
+    /**
+     * @returns {moment} Returns the locale moment object
+     */
+    getCurrentUserLocalTime() {
+        return DateUtils.getLocalMomentFromDatetime(
+            this.props.preferredLocale,
+            null,
+            this.props.timezone.selected,
+        );
+    }
+
+    /**
+     * @returns {string} Returns the timezone name in string, ie: GMT +07
+     */
+    getTimezoneName() {
+        // With non-GMT timezone, moment.zoneAbbr() will return the name of that timezone, so we can use it directly.
+        if (Number.isNaN(Number(this.state.currentUserLocalTime.zoneAbbr()))) {
+            return this.state.currentUserLocalTime.zoneAbbr();
+        }
+
+        // With GMT timezone, moment.zoneAbbr() will return a number, so we need to display it as GMT {abbreviations} format, ie: GMT +07
+        return `GMT ${this.state.currentUserLocalTime.zoneAbbr()}`;
+    }
+
+    /**
+     * Update the user's local time at the top of every minute
+     */
+    updateCurrentTime() {
+        if (this.timer) {
```

Comment:
I just tested this and I am trying to understand what's the use of this `if` block. 

Because by the time component updates, `setTimeout` is already called,  so there's no advantage of calling `clearTimeout` and making `this.timer = null`. You can just remove the whole if block.

I understand we need `clearTimeout(this.timer)` for `componentWillUnmount`, but definitely not needed inside `updateCurrentTime`.  do you have any strong reason to keep this?

Reply:
I have the same thought, but I also have a small concern about `componentDidUpdate` getting called because of prop update (i.e. you get a pusher event updating the `props.timezone`). In such case, a second `setTimeout` could get scheduled while the current one has not fired yet. Then, both of the `setTimeout` will schedule a new `setTimeout` each and they won't clear correctly during `componentWillUnmount`

Reply:
Got it. Thanks for clarifying @aldo-expensify. I wasn't aware if we could have pusher event updating the timezone.

Reply:
@hungvu193 You can ignore my comment here.

Reply:
> Got it. Thanks for clarifying @aldo-expensify. I wasn't aware if we could have pusher event updating the timezone.

Yes, normally what we try to do is.. if it is updating data, we push it so every client device that needs the info gets the update